opm/simulators/wells/BlackoilWellModel_impl.hpp

```diff
@@ -436,10 +436,10 @@ namespace Opm {
             // This is done only for producers, as injectors will only have a single
             // nonzero phase anyway.
             for (const auto& well : well_container_) {
-                const bool zero_target = well->stoppedOrZeroRateTarget(simulator_, this->wellState(), local_deferredLogger);
-                if (well->isProducer() && !zero_target) {
-                    well->updateWellStateRates(simulator_, this->wellState(), local_deferredLogger);
-                }
+                //const bool zero_target = well->stoppedOrZeroRateTarget(simulator_, this->wellState(), local_deferredLogger);
+                //if (well->isProducer()){// && !zero_target) {
+                    well->initializeWellState(simulator_, this->groupState(), this->wellState(), local_deferredLogger);
```

Comment:
Here, the function `initializeWellState()` will be used for both producers and injectors.  Needs to be careful.